Android - Fix crash on Jetpack login without WordPress

When logging into Jetpack without WordPress installed, blogging reminders migration crashes the app with a null cursor. This should be guarded along with the rest of the migration flows.

Reproduction steps:

1. Uninstall any Jetpack and WordPress apps.
2. Install Jetpack [release candidate 21.3-rc-3](https://github.com/wordpress-mobile/WordPress-Android/releases/tag/21.3-rc-3)
3. Log in to a WordPress.com account with sites
4. 💥 

<details>
<summary>Stack trace</summary>


```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.jetpack.android.beta, PID: 21935
    org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
        at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:537)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:519)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:511)
        at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7839)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.IllegalStateException: NullCursor(forEntity=BloggingReminders)
        at org.wordpress.android.bloggingreminders.resolver.BloggingRemindersResolver.trySyncBloggingReminders(BloggingRemindersResolver.kt:126)
        at org.wordpress.android.ui.main.WPMainActivity.onSiteChanged(WPMainActivity.java:1622)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:517)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:511) 
        at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.app.ActivityThread.main(ActivityThread.java:7839) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003) 
```
</details>

--- a/WordPress/src/main/java/org/wordpress/android/bloggingreminders/resolver/BloggingRemindersResolver.kt
+++ b/WordPress/src/main/java/org/wordpress/android/bloggingreminders/resolver/BloggingRemindersResolver.kt
@@ -12,6 +12,9 @@ import org.wordpress.android.fluxc.store.SiteStore
 import org.wordpress.android.localcontentmigration.LocalContentEntity.BloggingReminders
 import org.wordpress.android.localcontentmigration.LocalContentEntityData.BloggingRemindersData
 import org.wordpress.android.localcontentmigration.LocalMigrationContentResolver
+import org.wordpress.android.localcontentmigration.LocalMigrationResult.Companion.EmptyResult
+import org.wordpress.android.localcontentmigration.otherwise
+import org.wordpress.android.localcontentmigration.thenWith
 import org.wordpress.android.modules.APPLICATION_SCOPE
 import org.wordpress.android.ui.bloggingreminders.BloggingRemindersModelMapper
 import org.wordpress.android.ui.prefs.AppPrefsWrapper
@@ -35,14 +38,17 @@ class BloggingRemindersResolver @Inject constructor(
             onFailure()
             return
         }
-        val (reminders) = localMigrationContentResolver.getDataForEntityType<BloggingRemindersData>(BloggingReminders)
-        if (reminders.isNotEmpty()) {
-            val success = setBloggingReminders(reminders)
-            if (success) onSuccess() else onFailure()
-        } else {
-            bloggingRemindersSyncAnalyticsTracker.trackSuccess(0)
-            onSuccess()
-        }
+        localMigrationContentResolver.getResultForEntityType<BloggingRemindersData>(BloggingReminders)
+                .thenWith { (reminders) ->
+                    if (reminders.isNotEmpty()) {
+                        val success = setBloggingReminders(reminders)
+                        if (success) onSuccess() else onFailure()
+                    } else {
+                        bloggingRemindersSyncAnalyticsTracker.trackSuccess(0)
+                        onSuccess()
+                    }
+                    EmptyResult
+        }.otherwise { onFailure() }
     }
 
     @Suppress("ReturnCount")

--- a/WordPress/src/main/java/org/wordpress/android/localcontentmigration/LocalMigrationContentResolver.kt
+++ b/WordPress/src/main/java/org/wordpress/android/localcontentmigration/LocalMigrationContentResolver.kt
@@ -31,16 +31,6 @@ class LocalMigrationContentResolver @Inject constructor(
     @PublishedApi internal val wordPressPublicData: WordPressPublicData,
     @PublishedApi internal val queryResult: QueryResult,
 ){
-    inline fun <reified T : LocalContentEntityData> getDataForEntityType(
-        entityType: LocalContentEntity,
-        entityId: Int? = null
-    ) = getResultForEntityType<T>(entityType, entityId).let {
-        when (it) {
-            is Success -> it.value
-            is Failure -> error(it.error)
-        }
-    }
-
     @PublishedApi internal inline fun <reified T : LocalContentEntityData> Cursor.getValue() =
             queryResult.getValue<T>(this)
 

